domain/src/main/java/com/teamagam/gimelgimel/domain/messages/poller/PolledMessagesProcessor.java

```diff
@@ -148,6 +152,7 @@ private void mapEntityToMessage(BaseMessageGeo message) {
         }
 
         private void displayGeoEntity(GeoEntity geoEntity) {
+            mGeoEntitiesRepository.add(geoEntity);
```

Comment:
We've talked about it, you're right - we can refactor the code to allow drawing only through an interactor that will handle it by itself (and hide the DisplayRepository from classes outside its package).
That being said, we'll have to postpone it :<